### PR TITLE
DOCSP-34087-compatibility-updates

### PR DESCRIPTION
**Pull request by jason-price-mongodb**
## DESCRIPTION

Consolidated legacy shell content into one page, adjusted content for style, and used include files. Keep in mind the technical detail has not been changed from the existing approved content, just consolidated and adjusted.

## STAGING

https://preview-mongodbjasonpricemongodb.gatsbyjs.io/mongodb-shell/DOCSP-34087-compatibility-updates/reference/compatibility/#compatibility-changes-with-legacy-mongo-shell

https://preview-mongodbjasonpricemongodb.gatsbyjs.io/mongodb-shell/DOCSP-34087-compatibility-updates/reference/data-types

https://preview-mongodbjasonpricemongodb.gatsbyjs.io/mongodb-shell/DOCSP-34087-compatibility-updates/reference/ejson/stringify/#legacy-tojsononeline--

https://preview-mongodbjasonpricemongodb.gatsbyjs.io/mongodb-shell/DOCSP-34087-compatibility-updates/reference/methods

https://preview-mongodbjasonpricemongodb.gatsbyjs.io/mongodb-shell/DOCSP-34087-compatibility-updates/reference/options

https://preview-mongodbjasonpricemongodb.gatsbyjs.io/mongodb-shell/DOCSP-34087-compatibility-updates/run-commands/#command-exceptions

## JIRA

https://jira.mongodb.org/browse/DOCSP-34087

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=66cf8ef1c87522bd3a4e5fb1

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/includes/legacy/legacy-command-exceptions.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+For commands whose output includes ``{ ok: 0 }``, ``mongosh`` returns a
+consistent exception and omits the server raw output. The legacy
+``mongo`` shell returned output that varied for each command.
```
Comment (jmd-mongo) [inline source/includes/legacy/legacy-command-exceptions.rst:3]: [optional suggestion]

Consider updating statements about legacy mongo shell behavior to the present tense.

File: source/includes/legacy/legacy-shell-configuration.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+The legacy ``mongo`` shell used a configuration file named
```
Comment (jmd-mongo) [inline source/includes/legacy/legacy-shell-configuration.rst:1]: Same suggestion as above w/r/t present tense.

**Review by jmd-mongo — APPROVED** [review]
Thanks for these improvements, @jason-price-mongodb! I've left a suggestion for your consideration, but this one LGTM either way.

Thank you,
Joe

File: source/includes/legacy/legacy-command-exceptions.rst (hunk 1 above)
Comment (jason-price-mongodb) [reply source/includes/legacy/legacy-command-exceptions.rst]: The original content used past tense for the legacy shell, but will go ahead and change for now.

File: source/includes/legacy/legacy-shell-configuration.rst (hunk 2 above)
Comment (jason-price-mongodb) [reply source/includes/legacy/legacy-shell-configuration.rst]: See prev reply.

**Review by jason-price-mongodb — COMMENTED** [review]
Updates

**Review by addaleax — APPROVED** [review]

**Comment by jason-price-mongodb** [comment]
On Slack, Gaurab said it was okay to merge the content based on Anna's review.

**Pull request merged**